env0/resource_agent_project_assignment.go

```diff
@@ -0,0 +1,196 @@
+	AgentId   string `json:"agent_id"`
+	ProjectId string `json:"project_id"`
+}
+
+func GetAgentProjectAssignmentId(agentId string, projectId string) string {
+	return agentId + "_" + projectId
+}
+
+func (a *AgentProjectAssignment) GetId() string {
+	return GetAgentProjectAssignmentId(a.AgentId, a.ProjectId)
+}
+
+func resourceAgentProjectAssignment() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceAgentProjectAssignmentCreate,
+		ReadContext:   resourceAgentProjectAssignmentRead,
+		DeleteContext: resourceAgentProjectAssignmentDelete,
+
+		Importer: &schema.ResourceImporter{StateContext: resourceAgentProjectAssignmentImport},
+
+		Schema: map[string]*schema.Schema{
+			"agent_id": {
+				Type:        schema.TypeString,
+				Description: "id of the agent",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"project_id": {
+				Type:        schema.TypeString,
+				Description: "id of the project",
+				Required:    true,
+				ForceNew:    true,
+			},
+		},
+	}
+}
+
+func resourceAgentProjectAssignmentCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var newAssignment AgentProjectAssignment
+	if err := readResourceData(&newAssignment, d); err != nil {
+		return diag.Errorf("schema resource data deserialization failed: %v", err)
+	}
+
+	apaLock.Lock()
+	defer apaLock.Unlock()
+
+	apiClient := meta.(client.ApiClientInterface)
+	assignments, err := apiClient.ProjectsAgentsAssignments()
+	if err != nil {
+		return diag.Errorf("could not get project-agent assignments: %v", err)
+	}
+
+	for projectId, agentId := range assignments.ProjectsAgents {
+		if projectId == newAssignment.ProjectId && agentId == newAssignment.AgentId {
+			return diag.Errorf("assignment for project id %v and agent id %v already exist", projectId, agentId)
+		}
+	}
+
+	assignments.ProjectsAgents[newAssignment.ProjectId] = newAssignment.AgentId
+
+	if _, err := apiClient.AssignAgentsToProjects(assignments.ProjectsAgents); err != nil {
+		return diag.Errorf("could not update project-agent assignments: %v", err)
+	}
+
+	d.SetId(newAssignment.GetId())
+
+	return nil
+}
+
+func resourceAgentProjectAssignmentRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(client.ApiClientInterface)
+
+	apaLock.Lock()
+	defer apaLock.Unlock()
+
+	assignments, err := apiClient.ProjectsAgentsAssignments()
+	if err != nil {
+		return diag.Errorf("could not get project-agent assignments: %v", err)
+	}
+
+	var assignment *AgentProjectAssignment
+	for projectId, agentId := range assignments.ProjectsAgents {
+		if d.Id() == GetAgentProjectAssignmentId(agentId.(string), projectId) {
+			assignment = &AgentProjectAssignment{
+				AgentId:   agentId.(string),
+				ProjectId: projectId,
+			}
+			break
+		}
+	}
+
+	if assignment == nil {
+		log.Printf("[WARN] Drift Detected: Terraform will remove %s from state", d.Id())
+		d.SetId("")
+		return nil
+	}
```

Comment:
is there a way to test this as well? seems not covered

Reply:
@sagilaufer1992
https://github.com/env0/terraform-provider-env0/pull/325/commits/80a25e23a96c0a92a23c74a8cda3c6ab7d3d30e1